examples/autoscale/main.tf

```diff
@@ -170,77 +146,51 @@ module "autoscale" {
-  metadata = {
-    mgmt-interface-swap                  = "enable"
-    vmseries-bootstrap-gce-storagebucket = "my-google-bootstrap-bucket"
-    serial-port-enable                   = true
-    ssh-keys                             = "~/.ssh/vmseries-ssh-key.pub"
-  }
-  */
-  named_ports = [
-    {
-      name = "http"
-      port = "80"
-    },
+  depends_on = [
+    module.mgmt_cloud_nat
   ]
 }
 
-
-#-----------------------------------------------------------------------------------------------
-# Internal Network Balancer to distribute traffic to VM-Series trust interfaces
-#-----------------------------------------------------------------------------------------------
-/* 
-  The load balancers are not required for this example.  It is here to provide an example
-  of how to use the load balancer modules. 
-*/
-
+# Internal Network Load Balancer to distribute traffic to VM-Series trust interfaces
 module "intlb" {
   source = "../../modules/lb_internal/"
 
-  name                = "${local.prefix}intlb"
-  network             = module.vpc_trust.network_id
-  subnetwork          = module.vpc_trust.subnets_self_links[0]
-  all_ports           = true
-  backends            = module.autoscale.backends
+  name              = "${var.name_prefix}internal-lb"
+  network           = module.vpc_trust.network_id
+  subnetwork        = module.vpc_trust.subnets_self_links[0]
+  all_ports         = true
+  health_check_port = 80
+  backends = {
+    backend = module.autoscale.regional_instance_group_id
+  }
   allow_global_access = true
 }
 
-#-----------------------------------------------------------------------------------------------
-#  External Network Load Balancer to distribute traffic to VM-Series untrust interfaces
-#-----------------------------------------------------------------------------------------------
-
+# External Network Load Balancer to distribute traffic to VM-Series untrust interfaces
 module "extlb" {
   source = "../../modules/lb_external/"
 
-  name  = "${local.prefix}extlb"
-  rules = { ("${local.prefix}rule0") = { port_range = 80 } }
+  name  = "${var.name_prefix}external-lb"
+  rules = { ("${var.name_prefix}rule0") = { port_range = 80 } }
 
   health_check_http_port         = 80
   health_check_http_request_path = "/"
 }
 
-# -----------------------------------------------------------------------------------------------
-# Cloud Nat for the management interfaces.
-# -----------------------------------------------------------------------------------------------
-/* 
-  Cloud NAT is required in teh management network to provide connectivity to Cortex Data Lake
-  and to license the VM-Series from the PANW licensing servers.
-*/
-
+# Cloud NAT for the management interfaces. It is required in the management network to provide
+# connectivity to Cortex Data Lakeand to license the VM-Series from the PANW licensing servers.
 module "mgmt_cloud_nat" {
   source  = "terraform-google-modules/cloud-nat/google"
   version = "=1.2"
```

Comment:
Maybe we should bump this version a bit ? Right now the module is at version 4.0 : https://registry.terraform.io/modules/terraform-google-modules/cloud-nat/google/latest

Reply:
Out of scope, let's do it as part of refactor for NGFW Hub.